lib/install/client/install.go

```diff
@@ -146,12 +146,14 @@ type InstallerStrategy struct {
 	ConnectTimeout time.Duration
 }
 
-// isServiceFailed returns an error if the service has failed.
-func isServiceFailed(serviceName string) func() error {
+// shouldReconnectService returns an error if the service has failed or has been uninstalled
+func shouldReconnectService(serviceName string) func() error {
 	return func() error {
-		err := service.IsFailed(serviceName)
+		err := service.IsStatus(serviceName,
+			systemservice.ServiceStatusFailed,
+			systemservice.ServiceStatusUnknown)
```

Comment:
Added the unknown status to catch uninstall events since the dialer runs the connect loop separate from the service uninstall logic and this way will cancel connect attempts at the end of the completed operation faster.

Reply:
Would not this cause a sort of race condition issue where say a service is not being uninstalled, but rather not started yet? Not sure if this makes sense, I guess it depends on when "unknown" is actually being returned, just trying to think through various edge cases.

Reply:
Once the service has been installed, it cannot be in `unknown` state but I agree it is also very vague - I'll see if I can fetch more details to make a better judgement whether the service is still installed.

Reply:
On another thought there is not much merit in considering the additional status. I'll go for explicitly canceling the dialing loop when necessary with no changes to this logic.